syntax = "proto3";

package vacation.hotel;

import "shared/data_types/common.proto";

// ===== REQUEST =====

message HotelSearchRequest {
  vacation.common.Location location = 1;
  vacation.common.DateRange dates = 2;
  int32 guests = 3;
  int32 rooms = 4;
  repeated vacation.common.PreferenceType preferences = 5;
  
  // Filters
  int32 max_results = 6;
  double max_price_per_night = 7;
  double min_rating = 8;  // 0-5 scale
  repeated string amenities = 9;  // wifi, pool, gym, etc.
}

// ===== RESPONSE =====

message HotelSearchResponse {
  repeated HotelOption options = 1;
  SearchMetadata metadata = 2;
}

message HotelOption {
  string id = 1;
  
  // Basic info
  string name = 2;
  vacation.common.Location location = 3;
  double distance_to_center_km = 4;
  
  // Rating
  double rating = 5;  // 0-5 scale
  int32 review_count = 6;
  string rating_category = 7;  // e.g., "Excellent", "Very Good"
  
  // Pricing
  vacation.common.Money price_per_night = 8;
  vacation.common.Money total_price = 9;
  repeated Fee additional_fees = 10;
  
  // Room details
  RoomInfo room = 11;
  
  // Hotel features
  repeated string amenities = 12;
  string category = 13;  // budget, midscale, upscale, luxury
  int32 star_rating = 14;  // 1-5 stars
  
  // Scores
  vacation.common.ComponentScores scores = 15;
  
  // Booking
  string booking_url = 16;
  string provider = 17;
  bool available = 18;
  
  // Images
  repeated string image_urls = 19;
}

message RoomInfo {
  string type = 1;  // standard, deluxe, suite, etc.
  int32 beds = 2;
  string bed_type = 3;  // king, queen, twin, etc.
  int32 max_occupancy = 4;
  double size_sqm = 5;
  repeated string features = 6;  // balcony, view, etc.
}

message Fee {
  string name = 1;  // resort fee, parking, etc.
  vacation.common.Money amount = 2;
  bool mandatory = 3;
}

message SearchMetadata {
  int32 total_results = 1;
  string search_id = 2;
  string timestamp = 3;
  string data_source = 4;
}
